syft/frameworks/torch/mpc/securenn.py

```diff
@@ -321,37 +333,66 @@ def share_convert(a_sh):
     r_shares = r_sh.child
 
     alpha0 = (
-        (r_shares[workers[0].id] + r_shares[workers[1].id].copy().move(workers[0])) >= L
-    ).long()
+        (
+            (r_shares[workers[0].id] + r_shares[workers[1].id].copy().move(workers[0]))
+            > ((L - 1) // 2)
+        )
+    ).type(torch_dtype)
     alpha1 = alpha0.copy().move(workers[1])
     alpha = sy.MultiPointerTensor(children=[alpha0, alpha1])
 
-    u_sh = _shares_of_zero(1, L - 1, crypto_provider, *workers)
+    u_sh = _shares_of_zero(1, L - 1, "custom", crypto_provider, *workers)
 
     # 2)
     a_tilde_sh = a_sh + r_sh
     a_shares = a_sh.child
-    beta0 = ((a_shares[workers[0].id] + r_shares[workers[0].id]) >= L).long()
-    beta1 = ((a_shares[workers[1].id] + r_shares[workers[1].id]) >= L).long()
+    beta0 = (
+        ((a_shares[workers[0].id] + r_shares[workers[0].id]) > ((L - 1) // 2))
+        + ((a_shares[workers[0].id] + r_shares[workers[0].id]) < -(L // 2))
```

Comment:
Can't I just do this with memorize decorator on a field_max and field_min methods?

Reply:
We call this modulo method only for custom dtype which is only for internal use so I know it will only get 3 field values (2**32 -1, 2**64 -1, 67). Which can be memorized to speed it up .

Reply:
I agree! But as the  `(L - 1) // 2`  isn't that slow, you won't get substantial gain using memorize like this. I would suggest keeping those as attributes for the AST (well property to be exact) and compute them them only when required (possibly using memrize this time). That's not the thing which will bring huge speed up but it will help :)

Reply:
I added the property for min_value and max_value but I am not sure how to cache it into private variable after computing first time from there. Need help.